main_test.go

```diff
@@ -162,32 +217,58 @@ var _ = Describe("Main", func() {
 						ghttp.VerifyRequest("GET", "/v2/organizations/"+orgGuid),
 						ghttp.RespondWithJSONEncoded(http.StatusOK, testOrgResource(orgGuid)),
 					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v2/apps"),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, testAppResponse(afterApps)),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v2/spaces/"+spaceGuid),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, testSpaceResource(spaceGuid, orgGuid)),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v2/organizations/"+orgGuid),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, testOrgResource(orgGuid)),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v2/spaces/"+spaceGuid),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, testSpaceResource(spaceGuid, orgGuid)),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v2/organizations/"+orgGuid),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, testOrgResource(orgGuid)),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v2/spaces/"+spaceGuid),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, testSpaceResource(spaceGuid, orgGuid)),
+					),
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/v2/organizations/"+orgGuid),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, testOrgResource(orgGuid)),
+					),
 				)
 			})
 
 			It("should start three watchers and disconnect when requested", func() {
-				Expect(metricProc.updateApps()).To(Succeed())
-
-				var event *metrics.Stream
 
-				appEvents := map[string]*events.Envelope{}
-				for i := 0; i < len(apps); i++ {
-					Eventually(metricProc.msgChan).Should(Receive(&event))
-					appEvents[event.App.Guid] = event.Msg
-				}
+				Expect(metricProc.updateApps()).To(Succeed())
 
-				var connected func() int
 				for _, app := range apps {
-					connected = func() int {
-						return tcHandler.Connected(app.Guid)
+					guid := app.Guid
+					inMap := func() bool {
+						return metricProc.isWatched(guid)
 					}
+					Eventually(inMap).Should(Equal(true))
```

Comment:
I prefer to use built-in matchers where they exist: `Should(BeTrue())` instead of `Should(Equal(true))`

Reply:
🙌